zeppelin-web/src/app/notebook/paragraph/paragraph.controller.js

```diff
@@ -522,11 +523,20 @@ angular.module('zeppelinWebApp').controller('ParagraphCtrl', function($scope, $r
 
   $scope.aceChanged = function() {
     $scope.dirtyText = $scope.editor.getSession().getValue();
-    $scope.startSaveTimer();
 
-    $timeout(function() {
-      $scope.setParagraphMode($scope.editor.getSession(), $scope.dirtyText, $scope.editor.getCursorPosition());
-    });
+    if ($scope.dirtyText === undefined || $scope.originalText === undefined) {
+      return;
+    }
+
+    if ($scope.dirtyText.length !== $scope.originalText.length) {
+      $scope.isParagraphSaveAlready = true;
+      $scope.startSaveTimer();
+      var cursorValue = $scope.editor.getCursorPosition();
+      if (cursorValue.row === 0 && cursorValue <= 30) {
```

Comment:
AFAIK here, in case it's the first row (like `%something.somethig-else`) we change paragraph mode.
But why do we check `cursorValue <= 30` ?